clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareFetchRequestManager.java

```diff
@@ -201,6 +202,7 @@ private void handleShareFetchSuccess(Node fetchTarget, ShareSessionHandler.Share
                         partitionData,
                         requestVersion);
                 shareFetchBuffer.add(completedFetch);
+                shareFetchBuffer.handleAcknowledgementResponses(partition, Errors.forCode(partitionData.acknowledgeErrorCode()));
```

Comment:
We also need to set the acknowledge error codes if there is just a top-level error code.